### ISSUE TITLE
[$50] - Siderbar container has a performance bug! See and fix mapStateToProps

### ISSUE THREAD

**Issue by birdofpreyru** [topcoder-platform/community-app#649]
The problem is this line: https://github.com/topcoder-platform/community-app/blob/develop/src/shared/containers/challenge-listing/Sidebar.jsx#L129
Each time the `mapStateToProps(..)` function is re-evaluated, a new array will be created at this line and returned with the resulting object. Even if `state.tcCommunities.list.data` was not changed, React will think that props have changed and will trigger re-rendering of the underlying component. If should be fixed by returning from this function only the value from Redux state, and then appending the fake first element in the render function of the underlying component.

**Comment by arun1595** [comment]
@birdofpreyru I would like to work on this issue.

**Comment by andela-jomadoye** [comment]
I was able to take a shot at this issue, Today is my first day on this platform and codebase, so all suggestions and correction are welcome, this is the [PR](https://github.com/topcoder-platform/community-app/pull/707)

**Comment by birdofpreyru** [comment]
@andela-jomadoye cool! Always nice to see new faces at Topcoder :)
I'll review it tomorrow. One small note I have already: if you commit a fix for some ticket, is it handy to include the string `Fixes #XXX`, where `XXX` is the number of the issue it is supposed to fix. At GitHub this: (1) automatically creates a reference from the ticket to the commit/PR; (2) once the commit/PR is merged into `develop` branch, it automatically closes original ticket.

**Comment by andela-jomadoye** [comment]
Oh, Thanks for the tip @birdofpreyru
